great_tables/_helpers.py

```diff
@@ -518,6 +521,458 @@ def _get_font_stack(name: FontStackName = "system-ui", add_emoji: bool = True) -
+
+    from great_tables import GT, style, loc
+    import polars as pl
+
+    units_tbl = pl.DataFrame(
+        {
+            "rule": [
+                "unit with superscript",
+                "unit with subscript",
+                "unit with subscript and superscript",
+                "unit with subscript and superscript (using overstriking)",
+                "slashed-unit shorthand for a '-1' exponent",
+                "slashes between units normalized",
+                "multiple inline units, separating by a space",
+                "use of a number allowed with previous rules",
+                "use of 'x' preceding number to form scalar multiplier",
+                "hyphen is transformed to minus sign when preceding a unit",
+                "italicization of base unit",
+                "emboldening of base unit",
+                "italicizing and emboldening of base unit",
+                "styling of subscripts and superscripts",
+                "transformation of common units from ASCII to preferred form - 1",
+                "transformation of common units from ASCII to preferred form - 2",
+                "transformation of common units from ASCII to preferred form - 3",
+                "insertion of common symbols and Greek letters via `:[symbol name]:` - 1",
+                "insertion of common symbols and Greek letters via `:[symbol name]:` - 2",
+                "insertion of common symbols and Greek letters via `:[symbol name]:` - 3",
+                "use of chemical formulas via `%[chemical formula]%`",
+            ],
+            "input": [
+                "m^2",
+                "h_0",
+                "h_0^3",
+                "h[_0^3]",
+                "/s",
+                "t_0 / t_n",
+                "kg^2 m^-1",
+                "10^3 kg^2 m^-1",
+                "x10^3 kg^2 m^-1",
+                "-h^2",
+                "*m*^2",
+                "**m**^2",
+                "_**m**_^2",
+                "h_*0*^**3**",
+                "ug",
+                "um",
+                "degC",
+                ":angstrom:_2",
+                ":permille:C",
+                ":Sigma:^2 :delta:^-1 :alpha:^-1",
+                "g/L %C6H12O6%",
+            ],
+        }
+    ).with_columns(output=pl.col("input"))
+
+    (
+        GT(units_tbl)
+        .fmt_units(columns="output")
+        .tab_style(
+            style=style.text(font="courier"),
+            locations=loc.body(columns="input")
+        )
+    )
+    ```
+    """
+
+    # Get a list of raw tokens
+    tokens_list = _generate_tokens_list(units_notation=units_notation)
+
+    # Initialize a list to store the units
+    units_list = []
+
+    if len(tokens_list) == 0:
+        return UnitDefinitionList(units_list=[])
+
+    for i in range(len(tokens_list)):
+
+        tokens_list_i = tokens_list[i]
+
+        unit_subscript = None
+        sub_super_overstrike = False
+        chemical_formula = False
+        exponent = None
+
```

Comment:
Do you mind providing an example for these high-level cases in comments, so to queue people on what to look for in the regex code?

For example...

```python
        # Case: chemical formula
        #   * e.g. "%C6H12O6%"
        if re.match(r"^%.*%$", tokens_list_i) and len(tokens_list_i) > 2:
            chemical_formula = True

        # Case: subscript and superscript present, which requires an overstrike
        #   * e.g. "h[_0^3]"
        elif re.search(r".+?\[_.+?\^.+?\]", tokens_list_i):
            ...

        # Case: simple subscript and exponent
        #   * e.g. "h_0^3"
        elif re.search(r".+?_.+?\^.+?", tokens_list_i):
```

Having the case comments outside the blocks seems like it may help with reading a bit here, since regexes by themselves are a bit terse (so in this case the comment will be more prominent and hopefully give some hints to help with reading the regex).